### PR TITLE
added healthcheck output to marathon status

### PR DESCRIPTION
**Pull request by mjksmith**
Sample output
```
>>> from paasta_tools.marathon_serviceinit import perform_command
>>> perform_command('status', 'paasta_canary', 'mesosstage_main', 'mesosstage', 1, '/nail/etc/services')
State:      Running - Desired state: Started
Marathon:   Healthy - up with (5/5) instances. Status: Running.
  Marathon app ID: /paasta--canary.mesosstage--main.gitc19f9731.configdbd127ab
    App created: 2016-04-05 12:42:35.199000 (5 days ago)
    Tasks:
      Mesos Task ID                         Host deployed to           Deployed at what localtime         Health checks
      a57cc3df-0011-11e6-b964-0242a1bf1f9a  mesos2-uswest1cdevc:31194  2016-04-11T11:17 (16 minutes ago)  PASSING
      7bf354da-0010-11e6-a226-0242e21a17e4  mesos1-uswest1cdevc:31583  2016-04-11T11:09 (24 minutes ago)  PASSING
      b7f576c4-0011-11e6-b964-0242a1bf1f9a  mesos2-uswest1cdevc:31947  2016-04-11T11:17 (15 minutes ago)  PASSING
      9e42a30a-0011-11e6-b964-0242a1bf1f9a  mesos3-uswest1cdevc:31260  2016-04-11T11:17 (16 minutes ago)  PASSING
      9e42ca1c-0011-11e6-b964-0242a1bf1f9a  mesos2-uswest1cdevc:31959  2016-04-11T11:17 (16 minutes ago)  PASSING
Mesos:      Healthy - (5/5) tasks in the TASK_RUNNING state.
  Running Tasks:
    Mesos Task ID                         Host deployed to     Ram      CPU   Deployed at what localtime
    a57cc3df-0011-11e6-b964-0242a1bf1f9a  mesos2-uswest1cdevc  0/532MB  0.3%  2016-04-11T11:17 (16 minutes ago)
    9e42ca1c-0011-11e6-b964-0242a1bf1f9a  mesos2-uswest1cdevc  0/532MB  0.3%  2016-04-11T11:17 (16 minutes ago)
    9e42a30a-0011-11e6-b964-0242a1bf1f9a  mesos3-uswest1cdevc  0/532MB  0.4%  2016-04-11T11:17 (16 minutes ago)
    b7f576c4-0011-11e6-b964-0242a1bf1f9a  mesos2-uswest1cdevc  0/532MB  0.4%  2016-04-11T11:17 (15 minutes ago)
    7bf354da-0010-11e6-a226-0242e21a17e4  mesos1-uswest1cdevc  0/532MB  0.3%  2016-04-11T11:10 (23 minutes ago)
  Non-Running Tasks
    Mesos Task ID                         Host deployed to  Deployed at what localtime         Status
    a57c4eae-0011-11e6-b964-0242a1bf1f9a  Unknown           2016-04-11T11:17 (16 minutes ago)  TASK_FAILED
    abcc0f81-0011-11e6-b964-0242a1bf1f9a  Unknown           2016-04-11T11:17 (15 minutes ago)  TASK_FAILED
    b7639196-fffe-11e5-b964-0242a1bf1f9a  Unknown           2016-04-11T09:01 (2 hours ago)     TASK_FINISHED
    ae98d7ee-fff9-11e5-b964-0242a1bf1f9a  Unknown           2016-04-11T08:25 (3 hours ago)     TASK_FINISHED
    ae979f6d-fff9-11e5-b964-0242a1bf1f9a  Unknown           2016-04-11T08:25 (3 hours ago)     TASK_FAILED
    844ba08b-fff4-11e5-b964-0242a1bf1f9a  Unknown           2016-04-11T07:48 (3 hours ago)     TASK_FAILED
Smartstack:
  Haproxy Service Name: paasta_canary.mesosstage_main
  Backends:
      Name                             LastCheck        LastChange      Status
    uswest1-devc - Healthy - in haproxy with (5/5) total backends UP in this namespace.
      mesos2-uswest1cdevc:31194        L7OK/200 in 1ms  13 minutes ago  UP
      mesos1-uswest1cdevc:31583        L7OK/200 in 1ms  13 minutes ago  UP
      mesos2-uswest1cdevc:31947        L7OK/200 in 5ms  13 minutes ago  UP
      mesos3-uswest1cdevc:31260        L7OK/200 in 1ms  13 minutes ago  UP
      mesos2-uswest1cdevc:31959        L7OK/200 in 7ms  13 minutes ago  UP
      10-40-25-82-uswest1cdevc:31685   INI/ in ms       13 minutes ago  MAINT
      10-40-25-82-uswest1cdevc:31182   INI/ in ms       13 minutes ago  MAINT
      10-40-25-209-uswest1cdevc:31239  INI/ in ms       13 minutes ago  MAINT
      10-40-25-209-uswest1cdevc:31081  INI/ in ms       13 minutes ago  MAINT
```

### CODE REVIEW

**Comment by nhandler** [comment]
Looks like you have some test failures that need fixing.

File: paasta_tools/marathon_serviceinit.py [hunk 1]
```diff
@@ -138,22 +138,53 @@ def get_verbose_status_of_marathon_app(app):
...
-                local_deployed_datetime.strftime("%Y-%m-%dT%H:%M"),
-                humanize.naturaltime(local_deployed_datetime),
-            )
+            get_datestring_for_date(task.staged_at),
         ))
     output.append('\n'.join(["      %s" % line for line in format_table(rows)]))
+
+    output.append("    Healthchecks:")
+
+    rows = [("Mesos Task ID", "Status", "First Success", "Last Success", "Last Failure",
```
Comment (solarkennedy) [inline paasta_tools/marathon_serviceinit.py]: I don't think we need a whole other section just for healthchecks. Also I "think" that this is probably too much information. I'm think that customers really only care about the .alive, and pretty much nothing else. The thing is these healthchecks are running continuously, the timestamp is going to be always in the seconds (in theory)

Please append the healthcheck state to the existing marathon section that starts on line 150

File: paasta_tools/marathon_serviceinit.py [hunk 2]
```diff
@@ -138,22 +138,53 @@ def get_verbose_status_of_marathon_app(app):
     output.append("    App created: %s (%s)" % (str(create_datetime), humanize.naturaltime(create_datetime)))
     output.append("    Tasks:")
 
+    def get_datestring_for_date(date):
```
Comment (solarkennedy) [inline paasta_tools/marathon_serviceinit.py]: This looks like a nested function, did you intend it to be?

File: paasta_tools/marathon_serviceinit.py (hunk 1 above)
Comment (mjksmith) [inline paasta_tools/marathon_serviceinit.py]: @solarkennedy are there any services that have more than one healthcheck? Marathon has health check results as an array, so I assure that's the most recent result for each health check.

Service authors might also care about Last Success.

File: paasta_tools/marathon_serviceinit.py (hunk 2 above)
Comment (mjksmith) [inline paasta_tools/marathon_serviceinit.py]: @solarkennedy I did but it doesn't really need to be. I'll break it out.

Comment (mjksmith) [inline paasta_tools/marathon_serviceinit.py]: Actually if we're removing all the output except for 'Alive' for health checks we don't really need it.

**Comment by mjksmith** [comment]
@solarkennedy updated with the changes you suggested

File: paasta_tools/marathon_serviceinit.py [hunk 3]
```diff
@@ -138,20 +138,30 @@ def get_verbose_status_of_marathon_app(app):
...
     for task in app.tasks:
         local_deployed_datetime = datetime_from_utc_to_local(task.staged_at)
         if task.host is not None:
             hostname = "%s:%s" % (task.host.split(".")[0], task.ports[0])
         else:
             hostname = "Unknown"
+        health_check_results = [health_check.alive for health_check in task.health_check_results
+                                if health_check.alive is not None]
+        if not health_check_results:
+            health_check_status = PaastaColors.grey("UNKNOWN")
```
Comment (solarkennedy) [inline paasta_tools/marathon_serviceinit.py]: How about `N/A`? (as not every service has healthchecks and unknown sounds like something might be wrong

File: paasta_tools/marathon_serviceinit.py [hunk 4]
```diff
@@ -138,20 +138,30 @@ def get_verbose_status_of_marathon_app(app):
     output.append("    App created: %s (%s)" % (str(create_datetime), humanize.naturaltime(create_datetime)))
     output.append("    Tasks:")
 
-    rows = [("Mesos Task ID", "Host deployed to", "Deployed at what localtime")]
+    rows = [("Mesos Task ID", "Host deployed to", "Deployed at what localtime", "Health checks")]
```
Comment (solarkennedy) [inline paasta_tools/marathon_serviceinit.py]: Can you do "Health"
and then later do "Healthy / Unhealthy" to be consistent with the marathon dashboard?

**Comment by solarkennedy** [comment]
fix + ship

**Pull request merged**